yt_dlp/extractor/allstar.py

```diff
@@ -0,0 +1,264 @@
+import functools
+import json
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    OnDemandPagedList,
+    UnsupportedError,
+    int_or_none,
+    parse_qs,
+    str_or_none,
+    traverse_obj,
+    urljoin,
+)
+
+
+_FIELDS = '''
+    _id
+    clipImageThumb
+    clipLink
+    clipLinkHLS
+    clipTitle
+    createdDate
+    shareId
+    user { _id }
+    username
+    views'''
+
+_EXTRA_FIELDS = '''
+    clipLength
+    clipSizeBytes
+    game'''
+
+_QUERIES = {
+    'clip': '''query ($id: String!) {
+        video: getClip(clipIdentifier: $id) {
+            %s %s
+        }
+    }''' % (_FIELDS, _EXTRA_FIELDS),
+    'montage': '''query ($id: String!) {
+        video: getMontage(clipIdentifier: $id) {
+            %s
+        }
+    }''' % _FIELDS,
+    'clips': '''query ($page: Int!, $user: String!, $game: Int) {
+        videos: clips(search: createdDate, page: $page, user: $user, mobile: false, game: $game) {
+            data { %s %s }
+        }
+    }''' % (_FIELDS, _EXTRA_FIELDS),
+    'montages': '''query ($page: Int!, $user: String!) {
+        videos: montages(search: createdDate, page: $page, user: $user) {
+            data { %s }
+        }
+    }''' % _FIELDS,
+    'mobile clips': '''query ($page: Int!, $user: String!) {
+        videos: clips(search: createdDate, page: $page, user: $user, mobile: true) {
+            data { %s %s }
+        }
+    }''' % (_FIELDS, _EXTRA_FIELDS),
+}
+
+_PAGESIZE = 10
+
+
+def _media_url_or_none(path):
+    return urljoin('https://media.allstar.gg/', str_or_none(path))
+
+
+def _profile_url_or_none(path):
+    return urljoin('https:/allstar.gg/u/', str_or_none(path))
+
+
+def _parse_video_data(video_data):
+    return traverse_obj(video_data, {
+        'id': ('_id', {str_or_none}),
+        'display_id': ('shareId', {str_or_none}),
+        'title': ('clipTitle', {str_or_none}),
+        'url': ('clipLink', {_media_url_or_none}),
+        'thumbnail': ('clipImageThumb', {_media_url_or_none}),
+        'duration': ('clipLength', {int_or_none}),
+        'filesize': ('clipSizeBytes', {int_or_none}),
+        'timestamp': ('createdDate', {int_or_none}),
+        'uploader': ('username', {str_or_none}),
+        'uploader_id': ('user', '_id', {str_or_none}),
+        'uploader_url': ('user', '_id', {_profile_url_or_none}),
+        'view_count': ('views', {int_or_none}),
+        'categories': ('game', {str_or_none}),
+    })
+
+
+def _set_webpage_url(info_dict):
+    video_id = info_dict.get('id')
+    video_url = info_dict.get('url')
+
+    if video_url is None or video_id is None:
+        return info_dict
+
+    base_name = 'clip' if '/clips/' in video_url else 'montage'
+    info_dict['webpage_url'] = f'https://allstar.gg/{base_name}?{base_name}={video_id}'
+    info_dict['webpage_url_basename'] = base_name
```

Comment:
In case of use profile I need to change the `webpage_url` and make it point to the video url and also the `webpage_url_basename` needs to reflect that change and be `clip` or `montage`.
ex:
```
"webpage_url": "https://allstar.gg/clip?clip=62b9e39f40b69a1253878775",
"webpage_url_basename": "clip",
```

Reply:
Sure, but I don't see why `webpage_url_basename` should be returned in the info dict. You generally only return values as outlined in the [InfoExtractor base class](https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/common.py#L105).

Reply:
Never mind, I see what I missed here. Sorry.